spyderlib/widgets/arrayeditor.py

```diff
@@ -414,11 +414,14 @@ def _sel_to_text(self, cell_range):
         """Copy an array portion to a unicode string"""
         row_min, row_max, col_min, col_max = get_idx_rect(cell_range)
         _data = self.model().get_data()
-        output = io.StringIO()
+        if PY3:
+            output = io.BytesIO()
+        else:
+            output = io.StringIO()
         np.savetxt(output,
                   _data[row_min:row_max+1, col_min:col_max+1],
                   delimiter='\t')
-        contents = output.getvalue()
+        contents = output.getvalue().decode('utf-8')

```

Comment:
No - the functionality was fine in Python 2. `decode` doesn't break Python 2 (but is necessary on Python 3) so I left it in both code paths to avoid another if statement. It seems slightly cleaner to me like that but it'd be fine either way.